client/src/dashboard/RewritingRuleModal.js

```diff
@@ -17,14 +17,15 @@ import defaultRulesData from '../mock-api/listRules';
 import { FormLabel } from '@mui/material';
 import RuleGraph from './RuleGraph';
 
-const AddRewritingRule = NiceModal.create(({user_id}) => {
+const RewritingRuleModal = NiceModal.create(({user_id, rule=null}) => {
```

Comment:
Your PR should not include changes from another PR. Otherwise, your PR or the other PR cannot be merged due to conflicts.

The right way is modify on top of the current master branch, and create your PR based on those changes. If you want to apply changes from the other PR to your local, the right way is the following:
(1) Your local computer has branches: `master` and `manageApplicationTab`, the latter has changes you made;
(2) You `git fetch origin`, `git checkout <the branch name from David>`, then locally, you have three branches now.
(3) You go back to your branch, `git checkout manageApplicationTab`, copy another branch based on this branch: `git checkout -b manageApplicationTab2`;
(4) Rebase the new branch on top of David's branch: `git rebase <the branch name from David>`;
(5) Now you have four branches locally: `master`, `manageApplicationTab`, `<the branch name from David>`, and `manageApplicationTab2`, and the last one has both your changes in branch `manageApplicationTab` and  in branch `<the branch name from David>`.
(6) Now you want to submit a PR with only your changes, go back to `manageApplicationTab`, `git checkout manageApplicationTab`;
(7) Push this branch to Github: `git push` (might need other parameters to setup the up-stream, I forgot the exact command).
(8) Then create PR using the pushed branch `manageApplicationTab`.

Reply:
Thank you so much for the detailed instruction! I have cleaned up all the changes made by the other PR.